odoo/tools/sql.py

```diff
@@ -544,32 +544,77 @@ def check_index_exist(cr, indexname):
     assert index_exists(cr, indexname), f"{indexname} does not exist"
 
 
-def create_index(cr, indexname, tablename, expressions, method='btree', where=''):
-    """ Create the given index unless it exists. """
+def index_definition(cr, indexname):
+    """ Read the index definition from the database """
+    cr.execute(SQL("""
+        SELECT COALESCE(d.description, idx.indexdef) AS def
+        FROM pg_class c
+        JOIN pg_indexes idx ON c.relname = idx.indexname
+        LEFT JOIN pg_description d ON c.oid = d.objoid
+        WHERE c.relname = %s AND c.relkind = 'i'
+    """, indexname))
+    return cr.fetchone()[0] if cr.rowcount else None
+
+
+def create_index(
+    cr,
+    indexname,
+    tablename,
+    expressions,
+    method='btree',
+    where='',
+    *,
+    comment=None,
+    nulls_distinct=True,
+    unique=False
+):
+    """ Create the given index unless it exists.
+
+    :param cr: The cursor
+    :param indexname: The name of the index
+    :param tablename: The name of the table
+    :param method: The type of the index (default: btree)
+    :param where: WHERE clause for the index (default: '')
+    :param comment: The comment to set on the index
+    :param nulls_distinct: Whether nulls should be distinct for unique indexes
```

Comment:
Is it used ?

Reply:
Not today, it is here for completeness sake.
I saw some unique indexes with `coalesce(field, -1)` where this option does exactly what is needed and is more optimal.

Reply:
It is generally a bad idea to add something unused. Also the feature cannot be used anyway because we are in PostgreSQL 14 and it was added in 15.

Reply:
Removed